confdgnmi/testtool/OpenConfig/01__interfaces.robot

```
@@ -0,0 +1,195 @@
+    ...                OK response that does not include the read-only YANG "state" container.
+    [Tags]  config  container  negative
+    Set Datatype To    CONFIG
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include    state
+
+Get "OPERATIONAL" response includes "state" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does include the read/write YANG "state" container.
+    [Tags]  operational  container
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  state
+
+Get "OPERATIONAL" response does not include "config" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does not include the read-only YANG "config" container.
+    [Tags]  operational  container  negative
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include  config
+
+Get "ALL" response includes both "config" and "state" containers
+    [Documentation]    Verify that ``GetRequest`` with ``type=ALL`` parameter receives
+    ...                OK response that does include both read-write "config" and read-only "state" containers.
+    [Tags]  config  operational  container
+    Set Datatype To    ALL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  config
+    Check last updates include  state
+
+Get "config" response includes significant leaves
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does include standardized leaf names/items.
+    [Tags]  config  leaf
+    [Template]    Get interface "config" includes leaf
+    name
+    type
+    mtu
+    loopback-mode
+    description
```

These tests differ little bit too much from the rest - in the other ones you explicitly set datatype, invoke get, and do something with the result, here you do it all in one keyword.
Perhaps more importantly - should we really fail the test if `description` is not set? I'd say only `name` and `type` are required, the other ones... I doubt it.

---

I agree that value of this may be questionable depending on testing goals.
I assumed tester wants to check "compliance" with standard YANG, thus included all the leaves.
`[Template]` tests are run as separate "tests" and can succeed/fail independently on each other...
This implies that "failure" for some of the leaves like `description` may be just a warning flag that device deviates from standard YANG. We may need to address our stance on cases like this one. (@waitai ?)

---

Do we know what are significant leaves? It may be different for different customers/scenarios/devices. If yes, the is should be part of device config.

---

If user does not need some leaves he can skip specific tests and/or ignore their failure results. 
If user wants 100% OpenConfig compliance, presence of all the leaves is to be expected.

Failing test case does not mean that device is "bad", it just shows what was tested, and what succeeded/failed.
Test case helps to show what target gNMI server device is capable of, and listing negative but reasonably expectable use-cases is IMHO useful as well.

Making any extra things configurable raises knowledge requirement on the test runner/user. This can be problematic due to forcing user to dig into potentially non-interesting details for every other test-case.

---

There may be default config. But the main question is how you know the chosen leafs are significant?

---

that's up to each test user to either skip or ignore leaves from standard that are not "significant" for their use-case :)